src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpSyncNamespaceService.cs

```diff
@@ -0,0 +1,348 @@
+                            .WithAdditionalAnnotations(WarningAnnotation),
+                    externs: default, 
+                    usings: default,
+                    members: compilationUnit.Members);
+                return compilationUnit.WithMembers(new SyntaxList<MemberDeclarationSyntax>(targetNamespaceDecl));
+            }
+
+            // We should have a single member which is a namespace declaration in this compilation unit.
+            var namespaceDeclaration = compilationUnit.DescendantNodes().OfType<NamespaceDeclarationSyntax>().Single();
+
+            // Move everything to global namespace
+            if (targetNamespaceParts.Length == 1 && targetNamespaceParts[0].Length == 0)
+            {
+                var (namespaceOpeningTrivia, namespaceClosingTrivia) = 
+                    GetOpeningAndClosingTriviaOfNamespaceDeclaration(namespaceDeclaration);
+                var members = namespaceDeclaration.Members;
+                var eofToken = compilationUnit.EndOfFileToken
+                    .WithAdditionalAnnotations(WarningAnnotation);
+
+                // Try to preserve trivia from original namesapce declaration.
+                // If there's any member inside the declaration, we attach them to the 
+                // first and last member, otherwise, simply attach all to the EOF token.
+                if (members.Count > 0)
+                {
+                    var first = members.First();
+                    var firstWithTrivia = first.WithPrependedLeadingTrivia(namespaceOpeningTrivia);
+                    members = members.Replace(first, firstWithTrivia);
+
+                    var last = members.Last();
+                    var lastWithTrivia = last.WithAppendedTrailingTrivia(namespaceClosingTrivia);
+                    members = members.Replace(last, lastWithTrivia);
+                }
+                else
+                {
+                    eofToken = eofToken.WithPrependedLeadingTrivia(
+                        namespaceOpeningTrivia.Concat(namespaceClosingTrivia));
+                }
+
+                // Moving inner imports out of the namespace declaration can lead to a break in semantics.
+                // For example:
+                //
+                //  namespace A.B.C
+                //  {
+                //    using D.E.F;
+                //  }
+                //
+                //  The using of D.E.F is looked up iwith in the context of A.B.C first. If it's moved outside,
+                //  it may fail to resolve.
+
+                return compilationUnit.Update(
+                    compilationUnit.Externs.AddRange(namespaceDeclaration.Externs), 
+                    compilationUnit.Usings.AddRange(namespaceDeclaration.Usings), 
+                    compilationUnit.AttributeLists, 
+                    members,
+                    eofToken);
+            }
+
+            // Change namespace name
+            return compilationUnit.ReplaceNode(namespaceDeclaration, 
```

Comment:
is the reason you use syntax directly to update the tree rather than using syntax editor is because syntax editor doesn't have any benefit over just using syntax directly? just asking to see whether you have considered using the syntax editor.

Reply:
Didn't use because I don't need batching or tracking in this case, and it's a fairly simple change.